jps-plugin/src/org/intellij/erlang/jps/builder/ErlangBuilder.java

```diff
@@ -83,19 +80,23 @@ public void build(@NotNull ErlangTarget target,
                     @NotNull DirtyFilesHolder<ErlangSourceRootDescriptor, ErlangTarget> holder,
                     @NotNull BuildOutputConsumer outputConsumer,
                     @NotNull CompileContext context) throws ProjectBuildException, IOException {
-    LOG.debug(target.getPresentableName());
-    if (!holder.hasDirtyFiles() && !holder.hasRemovedFiles()) return;
 
     JpsModule module = target.getModule();
     JpsProject project = module.getProject();
     ErlangCompilerOptions compilerOptions = JpsErlangCompilerOptionsExtension.getOrCreateExtension(project).getOptions();
     if (compilerOptions.myUseRebarCompiler) return;
 
+    LOG.info("Starting build for " + target.getPresentableName());
     File sourceOutput = getBuildOutputDirectory(module, false, context);
     File testOutput = getBuildOutputDirectory(module, true, context);
 
-    buildSources(target, context, compilerOptions, holder, outputConsumer, sourceOutput, false);
-    buildSources(target, context, compilerOptions, holder, outputConsumer, testOutput, true);
+    if (holder.hasRemovedFiles()) {
+      LOG.debug("Remove invalid .beam and .app files.");
+      removeOutputFiles(holder.getRemovedFiles(target), sourceOutput, testOutput);
```

Comment:
Please, comment the method body, otherwise the comment is still relevant after changes.